bladeui/js/pages/referrals.js

```diff
@@ -0,0 +1,104 @@
+"use strict";
+
+const BaseClass = require("./baseClass");
+
+class Referrals extends BaseClass
+{
+  constructor(props)
+  {
+    super(props);
+  }
+
+  initListeners()
+  {
+    const linkBnt = document.getElementById("go-to-referal-form-send-btn");
+    const optionsDropDown = document.getElementById("open-select-options");
+    const chooseOption = document.getElementById("choose-option");
+    this.emailListTarget = document.getElementById("referrals-list-content");
+    this.renderReferralsList(10);
+
+    linkBnt.addEventListener("click", this.handleLinkBnt.bind(this));
+    optionsDropDown.addEventListener("click",
+      this.handleOpenOptions.bind(this));
+    chooseOption.addEventListener("click", this.handleSelectOption.bind(this));
+  }
+
+  handleLinkBnt()
+  {
+    super.handleChangeView("referralsFormView");
+  }
+
+  handleOpenPreviousView()
+  {
+    super.handleChangeView("refferalsMenuView");
+  }
+
+  handleOpenOptions()
+  {
+    const dropdown = document.getElementsByClassName("options-container")[0];
```

Comment:
you should not use classname to get single element, please use id

Reply:
also, this method will be called each time options select opened
so, every time you will access to the dom for this element. Get it once in initListeners